Support APP_PATH of config.properties in URL download link;

ansonliao opened the pull request:
@saikrishna321 
please advise.
1. Support APP_PATH of config.properties in URL download link;
2. Added functions to parse APK for getting package name, launch activity, to parse IPA for getting bundle ID;
3. Updated ExtentCucumberFormatter.java to support CI URL.

saikrishna321 commented:
@ansonliao thanks for the PR. Can you pls fix the styling issues CI is failing 
https://travis-ci.org/saikrishna321/AppiumTestDistribution/builds/156561942#L1963

ansonliao commented:
@saikrishna321 
I checked AppiumTestDistribution from my personal Github repo, and compile on my local Mac machine to install JAR to my local machine's maven repository in succeeded, the log is below.

```bash
 ~/Documents/app/git/AppiumTestDistribution >>master $ mvn clean install -DskipTests
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.github.saikrishna321:AppiumTestDistribution:jar:5.0.2-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: com.googlecode.plist:dd-plist:jar -> duplicate declaration of version 1.16 @ line 146, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 303, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building AppiumTestDistribution 5.0.2-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ AppiumTestDistribution ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ AppiumTestDistribution ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ AppiumTestDistribution ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 34 source files to /Users/ansonliao/Documents/app/git/AppiumTestDistribution/target/classes
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[3,16] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[3,16] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[3,16] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[22,13] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[22,41] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/android/AndroidUtils.java: /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/android/AndroidUtils.java uses or overrides a deprecated API.
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/utils/android/AndroidUtils.java: Recompile with -Xlint:deprecation for details.
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/manager/ParallelThread.java: Some input files use unchecked or unsafe operations.
[WARNING] /Users/ansonliao/Documents/app/git/AppiumTestDistribution/src/main/java/com/appium/manager/ParallelThread.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ AppiumTestDistribution ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 16 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ AppiumTestDistribution ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 27 source files to /Users/ansonliao/Documents/app/git/AppiumTestDistribution/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ AppiumTestDistribution ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ AppiumTestDistribution ---
[INFO] Building jar: /Users/ansonliao/Documents/app/git/AppiumTestDistribution/target/AppiumTestDistribution-5.0.2-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ AppiumTestDistribution ---
[INFO] Installing /Users/ansonliao/Documents/app/git/AppiumTestDistribution/target/AppiumTestDistribution-5.0.2-SNAPSHOT.jar to /Users/ansonliao/.m2/repository/com/github/saikrishna321/AppiumTestDistribution/5.0.2-SNAPSHOT/AppiumTestDistribution-5.0.2-SNAPSHOT.jar
[INFO] Installing /Users/ansonliao/Documents/app/git/AppiumTestDistribution/pom.xml to /Users/ansonliao/.m2/repository/com/github/saikrishna321/AppiumTestDistribution/5.0.2-SNAPSHOT/AppiumTestDistribution-5.0.2-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 16.125 s
[INFO] Finished at: 2016-09-01T10:21:03+08:00
[INFO] Final Memory: 25M/294M
[INFO] ------------------------------------------------------------------------
```

BTW, I forked AppiumTestDistribution to my repo and then commit the changes to my repo, will also push to original AppiumTestDistribution repo? I found that your travis-ci built the code from  original AppiumTestDistribution repo.

saikrishna321 commented:
@ansonliao mvn clean site install -Dmaven.test.skip=true

ansonliao commented:
@saikrishna321 
1.
I fixed style issue, please check my personal repo again (https://github.com/ansonliao/AppiumTestDistribution.git).

2.
For ExtentX enable, I found that run command `ExtentX="true" mvn clean test -Dtest=Runner` will be fail as errors such as `Unknown Maven lifecycle`, so I changed class `ExtentManager.java` some codes, and now enable ExtentX by commandline is `mvn clean test -DExtentX="true" -Dtest=Runner`, 
if your code is correct, please rollback.

saikrishna321 commented on src/main/java/com/appium/manager/AppiumParallelTest.java:
```diff
@@ -696,4 +736,69 @@ public String currentDateAndTime() {
+        }
+    }
+
+//    @Test
```
Can we remove the commented code

ansonliao commented:
@saikrishna321 
Hi Sai, I no idea how to fix the style issue as my local compile is succeeded, please check my local compile log before.

```bash
$ > mvn clean site install -Dmaven.test.skip=true
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.github.saikrishna321:AppiumTestDistribution:jar:5.1.0-SNAPSHOT
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: com.googlecode.plist:dd-plist:jar -> duplicate declaration of version 1.16 @ line 146, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 303, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building AppiumTestDistribution 5.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ AppiumTestDistribution ---
[INFO] Deleting /git/AppiumTestDistribution/target
[INFO] 
[INFO] --- maven-site-plugin:3.3:site (default-site) @ AppiumTestDistribution ---
[INFO] configuring report plugin org.apache.maven.plugins:maven-checkstyle-plugin:2.17
[INFO] configuring report plugin org.apache.maven.plugins:maven-jxr-plugin:2.3
[WARNING] Report plugin org.apache.maven.plugins:maven-project-info-reports-plugin has an empty version.
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[INFO] configuring report plugin org.apache.maven.plugins:maven-project-info-reports-plugin:2.9
[INFO] Relativizing decoration links with respect to project URL: https://github.com/saikrishna321/AppiumTestDistribution
[INFO] Rendering site with org.apache.maven.skins:maven-default-skin:jar:1.0 skin.
[INFO] Skipped "Source Xref" report, file "xref/index.html" already exists for the English version.
[INFO] Skipped "Test Source Xref" report, file "xref-test/index.html" already exists for the English version.
[INFO] Generating "Checkstyle" report    --- maven-checkstyle-plugin:2.17
[INFO] Starting audit...
Audit done.
[INFO] Generating "Source Xref" report    --- maven-jxr-plugin:2.3
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Generating "Test Source Xref" report    --- maven-jxr-plugin:2.3
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Generating "Dependencies" report    --- maven-project-info-reports-plugin:2.9
[WARNING] The repository url 'http://repository.codehaus.org' is invalid - Repository 'codehaus' will be blacklisted.
[WARNING] The repository url 'http://people.apache.org/repo/m2-snapshot-repository' is invalid - Repository 'apache.snapshots' will be blacklisted.
[INFO] Generating "Dependency Convergence" report    --- maven-project-info-reports-plugin:2.9
[WARNING] While downloading junit:junit-dep:4.11
  This artifact has been relocated to junit:junit:4.11.
  The artifact junit:junit does not contain Hamcrest anymore but declares a dependency to Hamcrest. Thus, junit:junit-dep has become obsolete.


[INFO] Generating "Dependency Information" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Distribution Management" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "About" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Licenses" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Plugin Management" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Plugins" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Team" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Source Code Management" report    --- maven-project-info-reports-plugin:2.9
[INFO] Generating "Summary" report    --- maven-project-info-reports-plugin:2.9
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ AppiumTestDistribution ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ AppiumTestDistribution ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 32 source files to /git/AppiumTestDistribution/target/classes
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[3,16] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[3,16] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[3,16] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[22,13] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/utils/ImageDecoding.java:[22,41] sun.misc.BASE64Encoder is internal proprietary API and may be removed in a future release
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/manager/ParallelThread.java: Some input files use unchecked or unsafe operations.
[WARNING] /git/AppiumTestDistribution/src/main/java/com/appium/manager/ParallelThread.java: Recompile with -Xlint:unchecked for details.
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ AppiumTestDistribution ---
[INFO] Not copying test resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ AppiumTestDistribution ---
[INFO] Not compiling test sources
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ AppiumTestDistribution ---
[INFO] Tests are skipped.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ AppiumTestDistribution ---
[INFO] Building jar: /git/AppiumTestDistribution/target/AppiumTestDistribution-5.1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ AppiumTestDistribution ---
[INFO] Installing /git/AppiumTestDistribution/target/AppiumTestDistribution-5.1.0-SNAPSHOT.jar to /Users/AnsonLiao/.m2/repository/com/github/saikrishna321/AppiumTestDistribution/5.1.0-SNAPSHOT/AppiumTestDistribution-5.1.0-SNAPSHOT.jar
[INFO] Installing /git/AppiumTestDistribution/pom.xml to /Users/AnsonLiao/.m2/repository/com/github/saikrishna321/AppiumTestDistribution/5.1.0-SNAPSHOT/AppiumTestDistribution-5.1.0-SNAPSHOT.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:10 min
[INFO] Finished at: 2016-09-16T22:18:28+08:00
[INFO] Final Memory: 54M/677M
[INFO] ------------------------------------------------------------------------
```

saikrishna321 commented:
@ansonliao can you pls fix this .. 

https://travis-ci.org/saikrishna321/AppiumTestDistribution/builds/171896397#L983

SrinivasanTarget commented on src/main/java/com/appium/manager/AppiumParallelTest.java:
```diff
@@ -786,6 +816,123 @@ public String currentDateAndTime() {
+
+    /**
+     * Added by Anson Liao
+     */
```
Can we remove signature everywhere?

saikrishna321 commented:
@ansonliao can you upstream and fix the conflicts as we have Extent version 3

saikrishna321 commented:
@ansonliao  can you squash all the commits to single one..

saikrishna321 commented:
@ansonliao can u pls squash all the commits
